### PR TITLE
Add /election/<election_id>/contest API

### PR DESCRIPTION
**Pull request by jonahkagan**
**Description**

Task: #309, #326 

Adds new endpoints for getting and creating/updating contests when setting up an audit:
- `GET /election/<election_id>/contest` - gets a list of all contests for the audit
- `PUT /election/<election_id>/contest` - sets the contests for the audit, overwriting any existing contests

Notably, this new API includes a `jurisdictionIds` field to set the contest universe.

The new endpoints use JSON schema to validate the contests.

**Testing**

Added a new test file with quite a few cases.

**Progress**

Ready for review. Left some open questions as PR comments.

### CODE REVIEW

File: arlo_server/models.py [hunk 1]
```diff
@@ -183,17 +186,32 @@ class ContestChoice(db.Model):
...
         nullable=False,
-    )
+    ),
+    db.Column(
+        "jurisdiction_id",
+        db.String(200),
+        db.ForeignKey("jurisdiction.id", primary_key=True, ondelete="cascade"),
+        nullable=False,
+    ),
+)
```
Comment (jonahkagan) [inline arlo_server/models.py]: This is the recommended way to set up the relationship according to the flask-sqlalchemy docs. I'm a bit concerned about performance, because we have to look up all the jurisdictions when we set the `jurisdiction_ids` field on the contests.

https://flask-sqlalchemy.palletsprojects.com/en/2.x/models/#many-to-many-relationships

File: arlo_server/contests.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        contest_id=contest_id,
+        name=json_contest_choice["name"],
+        num_votes=json_contest_choice["numVotes"],
+    )
+
+
+def deserialize_contest(json_contest, election_id):
+    jurisdictions = (
+        db.session.query(Jurisdiction)
+        .filter(Jurisdiction.id.in_(json_contest["jurisdictionIds"]))
```
Comment (benadida) [inline arlo_server/contests.py]: add a filter here that the jurisdiction belong to the `election_id`.

File: arlo_server/contests.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    return Contest(
+        election_id=election_id,
+        id=json_contest["id"],
+        name=json_contest["name"],
+        is_targeted=json_contest["isTargeted"],
+        choices=choices,
+        total_ballots_cast=json_contest["totalBallotsCast"],
+        num_winners=json_contest["numWinners"],
+        votes_allowed=json_contest["votesAllowed"],
+        jurisdictions=jurisdictions,
```
Comment (benadida) [inline arlo_server/contests.py]: this works for adding new relationships? Cool!

File: arlo_server/models.py (hunk 1 above)
Comment (benadida) [reply arlo_server/models.py]: not at all worried, that field will be auto-indexed and the lookups will be fast, plus you're already doing it in one query for all jurisdictions.  We'll want to review indexes at some point, but architecturally, this is sound.

Comment (benadida) [reply arlo_server/models.py]: actually, the thing to be potentially worried about is not the query, but the *inserts* into the mapping table. We'll almost certainly want to background this at some point, cause it could be ~1600 inserts in a single contest. But that doesn't have to be in this PR.

File: bgcompute.py [hunk 4]
```diff
@@ -4,13 +4,15 @@
...
-from arlo_server.models import RoundContest
+from arlo_server.models import RoundContest, Contest
 from arlo_server.routes import compute_sample_sizes
 
 
 def bgcompute():
-    # round contests that don't have sample_size_options
-    round_contests = RoundContest.query.filter_by(sample_size_options=None)
+    # Round contests for targeted contests that don't have sample_size_options
+    round_contests = RoundContest.query.filter_by(Contest.is_targeted).filter_by(
```
Comment (benadida) [inline bgcompute.py]: sneaking in an unrelated change? ;)

File: arlo_server/contests.py (hunk 3 above)
Comment (jonahkagan) [reply arlo_server/contests.py]: This is straight outta the docs

File: bgcompute.py (hunk 4 above)
Comment (jonahkagan) [reply bgcompute.py]: whoops, that was from my earlier poking around. thanks for catching, wasn't supposed to be in here

File: arlo_server/contests.py (hunk 2 above)
Comment (jonahkagan) [reply arlo_server/contests.py]: added

**Comment by jonahkagan** [comment]
Ok, a few things happened since last review:
- There were type errors, and I ended up figuring out how to type the db models to get around it, which is pretty cool!
- I added type annotations to all my new code
- The tests were failing due to the db model changes, so I made some fixes to existing code to fix those tests

File: arlo_server/routes.py [hunk 5]
```diff
@@ -464,7 +465,7 @@ def audit_status(election_id=None):
             {
                 "id": j.id,
                 "name": j.name,
-                "contests": [c.contest_id for c in j.contests],
+                "contests": [c.id for c in j.contests],
```
Comment (benadida) [inline arlo_server/routes.py]: wow, was this just wrong code prior to your fix?

Comment (jonahkagan) [reply arlo_server/routes.py]: No, it worked before. My database model change broke it
